### PR TITLE
REF: dont alter self in _validate_specification

### PR DESCRIPTION
**Pull request by jbrockmendel**
Besides being better-practice, this will make it easier to annotate.  Done separately bc im tracking down what appear to be incorrect annotations that need to be fixed in order to annotate these methods.

### CODE REVIEW

File: pandas/core/reshape/merge.py [hunk 1]
```diff
@@ -1323,25 +1321,27 @@ def _create_cross_configuration(
             cross_col,
         )
 
-    def _validate_specification(self) -> None:
+    def _validate_specification(self, left_on, right_on):
```
Comment (mroeschke) [inline pandas/core/reshape/merge.py:1324]: Nit: While you're here, this function could use a better name like `_validate_left_right_on`

Comment (jbrockmendel) [reply pandas/core/reshape/merge.py]: good idea, updated

**Review by mroeschke — APPROVED** [review]

**Comment by jbrockmendel** [comment]
good here?  follow-up on deck

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**